### PR TITLE
Add an admin section to the site

### LINKED ISSUE #8: Site-wide administration
This ticket describes the site-wide administration features that will be available to administrator users to manage the site, add new studies, and so on.

### Administrators
- [x] Can add a new study by filling in the basic details (*details TBC*)
- [x] Can add new users to the site, optionally assigning them roles like Research Manager or PI
- [x] Can edit the "Topics" field options
- [x] Can edit the "Study type" field options
- [x] Can edit the "Location" field options
- [x] Can edit the "Impact type" field options
- [x] Can edit the "Document type" field options

### PR DESCRIPTION
**Pull request by stevenday**
This installs ActiveAdmin and uses it to create a basic admin site for editing
all of the simple stuff MSF admins ant to be able to control on the site.

It turns out this has a lot more features than I first thought, and is a lot
more configurable than (say) Django's admin (at least as we normally use it).

I think need to run through this with them (and show @zarino) so that we can
get the right split between building custom admin pages/functions and getting
things for free from this.

Closes #8

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -136,6 +179,12 @@ GEM
       rake (>= 0.8.7)
       thor (>= 0.18.1, < 2.0)
     rake (10.4.2)
+    ransack (1.7.0)
+      actionpack (>= 3.0)
+      activerecord (>= 3.0)
+      activesupport (>= 3.0)
+      i18n
+      polyamorous (~> 1.2)
```
Comment (davea) [inline Gemfile.lock]: *sigh*

File: db/structure.sql [hunk 2]
```diff
@@ -915,224 +965,245 @@ ALTER TABLE ONLY users
...
--- Name: index_users_on_reset_password_token; Type: INDEX; Schema: public; Owner: -; Tablespace:
+-- Name: index_users_on_reset_password_token; Type: INDEX; Schema: public; Owner: -; Tablespace: 
 --
 
 CREATE UNIQUE INDEX index_users_on_reset_password_token ON users USING btree (reset_password_token);
 
 
 --
--- Name: unique_schema_migrations; Type: INDEX; Schema: public; Owner: -; Tablespace:
+-- Name: unique_schema_migrations; Type: INDEX; Schema: public; Owner: -; Tablespace:
```
Comment (davea) [inline db/structure.sql]: Loads of these trailing whitespace chars have crept in from somewhere, probably not worth doing anything about it unless they disappear again next time the file is updated?

Comment (stevenday) [inline db/structure.sql]: Hmm, weird - but yeah, that file will get replaced whenever we next do a db migration (which will be very soon), so it's probably not worth worrying too much about at the moment.

File: app/admin/dashboard.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+ActiveAdmin.register_page "Dashboard" do
+  menu priority: 1, label: proc { I18n.t("active_admin.dashboard") }
+
+  content title: proc { I18n.t("active_admin.dashboard") } do
+    columns do
+      column do
+        panel "Studies over 12 months old that haven't completed" do
```
Comment (davea) [inline app/admin/dashboard.rb]: Hot damn this is nice. Django has some catching up to do :wink:

File: db/structure.sql (hunk 2 above)
Comment (davea) [inline db/structure.sql]: What I meant really it's not worth doing anything about unless it generates noise every time we PR something with changes to the whitespace in that file.

**Comment by davea** [comment]
As you mentioned in #8 there are a few things to be done to improve the user-friendliness of this, but overall it works nicely and looks good. Worth adding any tests at this point, or does it make more sense to do those later when the design/implementation is further along? 

:+1:

**Pull request closed without merging**